### PR TITLE
Persist Config file while spawning new instance

### PR DESCRIPTION
**Pull request by RohanPoojary**
PR to address issue: [4187](https://github.com/alacritty/alacritty/issues/4187).
This is my first contribution, please let me know if this isn't the expected solution.

I have tested it on my macOS(Mojave) and the new window uses the expected config file.

### CODE REVIEW

File: alacritty/src/event.rs [hunk 1]
```diff
@@ -314,9 +314,15 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
...
 
-            fs::read_link(link_path)
+            let mut args = fs::read_link(link_path)
                 .map(|path| vec!["--working-directory".into(), path])
-                .unwrap_or_default()
+                .unwrap_or_default();
+
+            // Adding currently loaded config-file in arguments.
+            args.push("--config-file".into());
+            args.extend(self.config.ui_config.config_paths.clone());
```
Comment (chrisduerr) [inline alacritty/src/event.rs:323]: We'd want to load all current arguments, not just config file. So we should lookup the current cmdline and just append it. That could be a bit of a problem if the working directory parameter already exists, but in general it is the most sensible solution.

**Review by chrisduerr — CHANGES_REQUESTED** [review]

File: alacritty/src/event.rs (hunk 1 above)
Comment (RohanPoojary) [reply alacritty/src/event.rs:323]: Agreed. Will append the arguments by reading it from command line arguments.
Also for windows the args variable expects it to be `Vec<String>` while on Unix its `Vec<OsStr>` is the difference expected?

Comment (RohanPoojary) [reply alacritty/src/event.rs]: @chrisduerr any remarks?

**Review by chrisduerr — APPROVED** [review]
Since I wasn't entirely sure how to best approach this, I've decided to implement the remaining stuff myself. Existing working directories should now be filtered out and all other CLI parameters are still passed correctly.

Since I've wrote a chunk of this code, @kchibisov should probably review it.

**Review by kchibisov — CHANGES_REQUESTED** [review]
It's not implemented for Windows however nothing stops us from doing so.

File: alacritty/src/event.rs [hunk 2]
```diff
@@ -299,10 +299,13 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
     }
 
     fn spawn_new_instance(&mut self) {
-        let alacritty = env::args().next().unwrap();
+        let mut env_args = env::args();
+        let alacritty = env_args.next().unwrap();
+
+        let working_directory_set;
```
Comment (kchibisov) [inline alacritty/src/event.rs:305]: The changes look fine, however I wonder whether we can return set working directory by returning a tuple in a place where you set `args`. so it'll be `let (mut args, working_directory_set) = { (args, false) };` That should save a bunch of lines.

Comment (chrisduerr) [reply alacritty/src/event.rs:305]: Actually there's a far simpler solution, I was just being silly.

File: alacritty/src/event.rs [hunk 3]
```diff
@@ -314,12 +315,29 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
...
         };
+
         #[cfg(not(unix))]
-        let args: Vec<String> = Vec::new();
+        let mut args: Vec<PathBuf> = Vec::new();
+
+        let working_directory_set = !args.is_empty();
+
+        // Reuse the arguments passed to Alacritty for the new instance.
+        while let Some(arg) = env_args.next() {
```
Comment (RohanPoojary) [inline alacritty/src/event.rs:332]: I think we can just use `filter` method here.
`
        args.extend(
            env_args
                .filter(|x| !working_directory_set || x != "--working-directory")
                .map(|x| x.into()),
        );

`

Comment (chrisduerr) [reply alacritty/src/event.rs:332]: We can't, look at what the code does. It doesn't just filter one element.

Comment (RohanPoojary) [reply alacritty/src/event.rs:332]: Alright, So this will be a place to filter out other arguments as well?

Comment (RohanPoojary) [reply alacritty/src/event.rs:332]: Oh right, my bad

**Review by kchibisov — APPROVED** [review]

**Pull request merged**